pkg/services/sqlstore/migrations/dashboard_public_mig.go

```diff
@@ -0,0 +1,52 @@
+package migrations
+
+import (
+	. "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addPublicDashboardMigration(mg *Migrator) {
```

Comment:
Can we add a condition so these migrations are only run when the feature is enabled? this was missing last time I checked. (It would have made all these migrations sort of unnecessary)

Reply:
I believe it was suggested (can't recall who) to not do that. @jalevin I think you have more context into why.

Reply:
Here's the discussion thread related to public dashboards. Unfortunately, this first migration has already been rolled out so the table exists.

https://github.com/grafana/grafana/pull/49131#issuecomment-1145456952

Reply:
eeep.  This is what I get now:

```
DEBUG[06-22|15:13:42] Executing migration condition SQL        logger=migrator id="create index UQE_dashboard_public_config_access_token - v1" sql="SELECT 1 FROM `sqlite_master` WHERE `type`='index' AND `tbl_name`=? AND `name`=?" args="[dashboard_public_config UQE_dashboard_public_config_access_token]"
DEBUG[06-22|15:13:42] Executing sql migration                  logger=migrator id="create index UQE_dashboard_public_config_access_token - v1" sql="CREATE UNIQUE INDEX `UQE_dashboard_public_config_access_token` ON `dashboard_public_config` (`access_token`);"
ERROR[06-22|15:13:42] Executing migration failed               logger=migrator id="create index UQE_dashboard_public_config_access_token - v1" error="no such column: access_token"
ERROR[06-22|15:13:42] Exec failed                              logger=migrator error="no such column: access_token" sql="CREATE UNIQUE INDEX `UQE_dashboard_public_config_access_token` ON `dashboard_public_config` (`access_token`);"
Failed to start grafana. error: migration failed (id = create index UQE_dashboard_public_config_access_token - v1): no such column: access_token
migration failed (id = create index UQE_dashboard_public_config_access_token - v1): no such column: access_token
[Bra] 06-22 15:13:42 [ WARN] Fail to execute command: ./bin/grafana-server [-packaging=dev cfg:app_mode=development] - exit status 1
```